### PR TITLE
Ensure sqlite folder is removed after clear cache / hard reset

### LINKED ISSUE #1554: rimraf the sqlite folder doesn't work... on Windows
The clear cache is still not OK on Windows, even with latest fixes (1.2.0).
We need to figure out why rimraf doesn't do the job.

### PR DESCRIPTION
**Pull request by meriadec**
closes #1554

Clean cache / hard reset has been reported (& reproduced) failing to remove the `<user-data>/sqlite` folder on Windows (high chance of folder being "hold" by some process, even if we kill libcore before..) and we get some EPERM and ENOTEMPTY (relatable: https://github.com/isaacs/rimraf/issues/72).

Fallback here is to detect this behaviour, and provide a workaround: open the user data folder then quit the app, explaining the user he has to delete manually.

Additionally for support guys, here is a video of the process (@dasilvarosa feel free to discuss the wording):

**DEMO:**

[kbwuldq](https://www.youtube.com/watch?v=7wzOY6Hn0oQ)

### CODE REVIEW

File: src/components/SettingsPage/CleanButton.js [hunk 1]
```diff
@@ -20,32 +21,35 @@ type Props = {
...
+  closeFallback = () => this.setState({ fallbackOpened: false })
 
   action = async () => {
     if (this.state.isLoading) return
     try {
       this.setState({ isLoading: true })
       await softReset({ cleanAccountsCache: this.props.cleanAccountsCache })
-    } finally {
-      this.setState({ isLoading: false })
+    } catch (err) {
```
Comment (gre) [inline src/components/SettingsPage/CleanButton.js]: `logger.error` it. (we should never silent errors for logs)

File: src/components/SettingsPage/ResetButton.js [hunk 2]
```diff
@@ -18,30 +19,34 @@ type Props = {
...
+  closeFallback = () => this.setState({ fallbackOpened: false })
 
   action = async () => {
     this.setState({ pending: true })
     try {
       await hardReset()
       remote.getCurrentWindow().webContents.reloadIgnoringCache()
     } catch (err) {
-      this.setState({ pending: false })
+      this.setState({ pending: false, fallbackOpened: true })
```
Comment (gre) [inline src/components/SettingsPage/ResetButton.js]: same. even tho it was already ignored !

File: src/components/SettingsPage/ResetFallbackModal.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        centered
+        isOpened={isOpened}
+        onConfirm={openUserDataFolderAndQuit}
+        onClose={onClose}
+        onReject={onClose}
+        confirmText={'Open folder'}
+        title="Couldnt remove app files"
+        desc={
+          <div>
+            <p>{'Cache folder couldnt be deleted. You will have to delete it manually.'}</p>
```
Comment (gre) [inline src/components/SettingsPage/ResetFallbackModal.js]: we need to have all these texts put in wording.

File: src/helpers/reset.js [hunk 4]
```diff
@@ -24,14 +30,19 @@ export async function hardReset() {
...
 export async function softReset({ cleanAccountsCache }: *) {
   cleanAccountsCache()
   await delay(500)
   await db.cleanCache()
-  resetLibcoreDatabase()
+  await resetLibcoreDatabase()
   reload()
 }
+
+export async function openUserDataFolderAndQuit() {
```
Comment (gre) [inline src/helpers/reset.js]: great idea!!!

**Comment by Arnaud97234** [comment]
No regression on macos and Ubuntu, sqlite folder is removed without need for user to delete the folder manually (even when application is syncing).

On Windows if "Clear cache" or "Reset Ledger Live" action is made just after sync ends, sqlite folder is removed without need for user to delete it manually.
If sqlite folder can't be removed automatically, the user will have to remove it manually.
A popup is displayed, on click on "Open folder" button app folder opens and Ledger Live process is killed. After removing sqlite folder, it's necessary to manually restart Ledger Live application.

File: static/i18n/en/app.json [hunk 5]
```diff
@@ -406,6 +406,13 @@
       "title": "Clear cache",
       "desc": "Clearing the Ledger Live cache forces network resynchronization. Your settings and accounts are not affected. The private keys to access your crypto assets in the blockchain remain secure on your Ledger device and on your Recovery sheet."
     },
+    "resetFallbackModal": {
+      "part1": "Cache folder couldnt be deleted. You will have to delete it manually.",
```
Comment (dasilvarosa) [inline static/i18n/en/app.json]: "part1": "Could not delete cache folder. Please delete the folder manually:",

File: static/i18n/en/app.json [hunk 6]
```diff
@@ -406,6 +406,13 @@
...
+      "part2": "Click on \"Open folder\", then the ",
```
Comment (dasilvarosa) [inline static/i18n/en/app.json]: "part2": "Click the Open folder button, the ",

File: static/i18n/en/app.json [hunk 7]
```diff
@@ -406,6 +406,13 @@
...
+      "part3": "app will close",
+      "part4": ", and you will have to delete the \"sqlite\" folder.",
```
Comment (dasilvarosa) [inline static/i18n/en/app.json]: "part4": ", make sure to delete the \"sqlite\" folder.",

File: static/i18n/en/app.json [hunk 8]
```diff
@@ -406,6 +406,13 @@
...
+      "part5": "After that, you can restart the app."
```
Comment (dasilvarosa) [inline static/i18n/en/app.json]: "part5": "Then you can restart the app normally."

File: src/components/SettingsPage/ResetFallbackModal.js [hunk 9]
```diff
@@ -24,18 +26,20 @@ class ResetFallbackModal extends PureComponent<Props> {
         title="Couldnt remove app files"
```
Comment (dasilvarosa) [inline src/components/SettingsPage/ResetFallbackModal.js]: title="User action required"

Comment (meriadec) [reply src/components/SettingsPage/ResetFallbackModal.js]: +1

**Comment by meriadec** [comment]
Updated !

**Comment by meriadec** [comment]
Updated !

2018-10-17_515x300

**Pull request merged**